cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1095 @@
+	pathBitfinex      = "https://docs.bitfinex.com/docs/changelog"
+	pathBitmex        = "https://www.bitmex.com/static/md/en-US/apiChangelog"
+	pathANX           = "https://anxv3.docs.apiary.io/"
+	pathPoloniex      = "https://docs.poloniex.com/#changelog"
+	pathIbBit         = "https://api.itbit.com/docs"
+	pathBTCMarkets    = "https://api.btcmarkets.net/openapi/info/index.yaml"
+	pathEXMO          = "https://exmo.com/en/api/"
+	pathBitstamp      = "https://www.bitstamp.net/api/"
+	pathHitBTC        = "https://api.hitbtc.com/"
+	pathBitflyer      = "https://lightning.bitflyer.com/docs?lang=en"
+	pathLakeBTC       = "https://www.lakebtc.com/s/api_v2"
+	pathKraken        = "https://www.kraken.com/features/api"
+	pathAlphaPoint    = "https://alphapoint.github.io/slate/#introduction"
+	pathGetAllLists   = "https://api.trello.com/1/boards/%s/lists?cards=none&card_fields=all&filter=open&fields=all&key=%s&token=%s"
+	pathNewCard       = "https://api.trello.com/1/cards?%s&key=%s&token=%s"
+	pathChecklists    = "https://api.trello.com/1/checklists/%s/checkItems?%s&key=%s&token=%s"
+	apiKey            = ""
+	apiToken          = ""
+	updateChecklistID = "5dfc5a5377835d0ba025787a"
+)
+
+var verbose bool
+
+func main() {
+	flag.BoolVar(&verbose, "verbose", false, "Increases logging verbosity for API Update Checker")
+	flag.Parse()
+	updates, err := CheckUpdates(jsonFile)
+	if err != nil {
+		log.Println(err)
+	}
+	log.Println(updates)
+}
+
+// GetSha gets the sha of the latest commit
+func GetSha(repoPath string) (ShaResponse, error) {
+	var resp ShaResponse
+	finalPath := fmt.Sprintf(githubPath, repoPath)
+	if verbose {
+		log.Println(finalPath)
+	}
+	return resp, common.SendHTTPGetRequest(finalPath, true, false, &resp)
+}
+
+// CheckExistingExchanges checks if the given exchange exists
+func CheckExistingExchanges(fileName, exchName string) ([]ExchangeInfo, bool, error) {
+	var data []ExchangeInfo
+	var err error
+	data, err = ReadFileData(fileName)
+	if err != nil {
+		return data, false, err
+	}
+	var resp bool
+	for x := range data {
+		if data[x].Name == exchName {
+			resp = true
+			break
+		}
+	}
+	return data, resp, nil
+}
+
+// ReadFileData reads the file data for the json file
+func ReadFileData(fileName string) ([]ExchangeInfo, error) {
+	jsonFile, err := os.Open(fileName)
+	if err != nil {
+		return nil, err
+	}
+	defer jsonFile.Close()
+	byteValue, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		return nil, err
+	}
+	var data []ExchangeInfo
+	err = json.Unmarshal(byteValue, &data)
+	if err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+// CheckUpdates checks Updates.json for all the existing exchanges
+func CheckUpdates(fileName string) ([]string, error) {
+	var resp []string
+	data, err := ReadFileData(fileName)
+	if err != nil {
+		return resp, err
+	}
+	for x := range data {
+		switch data[x].CheckType {
+		case github:
+			sha, err := GetSha(data[x].Data.GitHubData.Repo)
+			if err != nil {
+				return resp, err
```

Comment:
It would be nice to individually report errors instead of returning an err, so the APIs which work can still be reported